src/frontend/js/utils/function.tsx

```diff
@@ -0,0 +1,23 @@
+export const calcDepth = (depth: number) => {
```

Comment:
```suggestion
export const calcTopicItemDepth = (depth: number) => {
```

And I would name this file something like `topics.tsx` just to show what kind of utility it contains